From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pravalika Gurram <pravalika.gurram@intel.com>
Date: Sun, 21 Jun 2026 21:36:06 +0530
Subject: drm/ttm: guard NR_GPU_ACTIVE/NR_GPU_RECLAIM accounting

The NR_GPU_ACTIVE and NR_GPU_RECLAIM VM stats are not present in older
kernels. Guard the related page state accounting calls to avoid build
failures when backporting to kernels that lack these stats.

Reference:
	ae80122f3896 
	drm/ttm: use gpu mm stats to track gpu memory allocations.

Signed-off-by: Pravalika Gurram <pravalika.gurram@intel.com>
---
 drivers/gpu/drm/ttm/ttm_pool.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/gpu/drm/ttm/ttm_pool.c b/drivers/gpu/drm/ttm/ttm_pool.c
index 716ffb0..22b3db6 100644
--- a/drivers/gpu/drm/ttm/ttm_pool.c
+++ b/drivers/gpu/drm/ttm/ttm_pool.c
@@ -185,7 +185,9 @@ static struct page *ttm_pool_alloc_page(struct ttm_pool *pool, gfp_t gfp_flags,
 		p = alloc_pages_node(pool->nid, gfp_flags, order);
 		if (p) {
 			p->private = order;
+#ifndef BPM_NR_GPU_STATS_NOT_PRESENT
 			mod_lruvec_page_state(p, NR_GPU_ACTIVE, 1 << order);
+#endif
 		}
 		return p;
 	}
@@ -222,8 +224,10 @@ error_free:
 static void __free_pages_gpu_account(struct page *p, unsigned int order,
 				     bool reclaim)
 {
+#ifndef BPM_NR_GPU_STATS_NOT_PRESENT
 	mod_lruvec_page_state(p, reclaim ? NR_GPU_RECLAIM : NR_GPU_ACTIVE,
 			      -(1 << order));
+#endif
 	__free_pages(p, order);
 }
 
@@ -334,8 +338,10 @@ static void ttm_pool_type_give(struct ttm_pool_type *pt, struct page *p)
 	rcu_read_unlock();
 
 	atomic_long_add(num_pages, &allocated_pages[nid]);
+#ifndef BPM_NR_GPU_STATS_NOT_PRESENT
 	mod_lruvec_page_state(p, NR_GPU_ACTIVE, -num_pages);
 	mod_lruvec_page_state(p, NR_GPU_RECLAIM, num_pages);
+#endif
 }
 
 static enum lru_status take_one_from_lru(struct list_head *item,
@@ -360,8 +366,10 @@ static struct page *ttm_pool_type_take(struct ttm_pool_type *pt, int nid)
 	ret = list_lru_walk_node(&pt->pages, nid, take_one_from_lru, (void *)&p, &nr_to_walk);
 	if (ret == 1 && p) {
 		atomic_long_sub(1 << pt->order, &allocated_pages[nid]);
+#ifndef BPM_NR_GPU_STATS_NOT_PRESENT
 		mod_lruvec_page_state(p, NR_GPU_ACTIVE, (1 << pt->order));
 		mod_lruvec_page_state(p, NR_GPU_RECLAIM, -(1 << pt->order));
+#endif
 	}
 	return p;
 }
-- 
2.43.0

